crates/wit-component/tests/components/imports/component.wat

```diff
@@ -0,0 +1,166 @@
+(component
+  (type (;0;) (func (param "x" string)))
+  (type (;1;) (record (field "a" u8)))
+  (type (;2;) (func (param "x" (type 1))))
+  (type (;3;) 
+    (instance
+      (alias outer 1 (type (;0;) 0))
+      (export "bar1" (type 0))
+      (alias outer 1 (type (;1;) 1))
+      (export "x" (type 1))
+      (alias outer 1 (type (;2;) 2))
+      (export "bar2" (type 2))
+    )
+  )
+  (type (;4;) (list string))
+  (type (;5;) (func (param "x" (type 4))))
+  (type (;6;) (func))
+  (type (;7;) s8)
```

Comment:
This is because the interface definition uses a type alias (`type x = u8`).

Encoding it this way allows us to recreate the `wit` definition with the alias rather than inserting `u8` everywhere inline.

Reply:
Regardless of how we encode the naming of types (to be followed-up on the component model spec issue you've filed), I think we'll likely still want a way to encode primitive types like this so we can differentiate type aliases in function signatures; i.e. it's probably easier to name an aliased type index once than name every use in every function signature as an alias.

Resolving this for now and if we desire a different way to handle type aliases we can discuss it at the spec.